source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,185 @@
+===========================
+Update Arrays in a Document
+===========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to update arrays in a document.
+
+To update an array, you need to specify an update and the array elements
+you want to update to an update operation. 
+
+Sample Document
+~~~~~~~~~~~~~~~
+
+The following sections feature examples on how to update this sample
+document: 
+
+.. code-block:: json
+
+    { "_id": 1, "color": "green", "qty": [8, 12, 18] }
+
+The examples on this page use the ``findOneAndUpdate()`` method of the
+``MongoCollection`` class to retrieve and update our document. Each
+example uses an instance of ``FindOneAndUpdateOptions`` to configure the
+method to return the document  after the update occurs. For more
+information on the ``findOneAndUpdate()`` method, see our Compound
+Operations guide <TODO>. 
+
+Specifying an Update
+--------------------
+
+To specify an update, use an ``Updates`` builder. The ``Updates``
+builder provides static utility methods to construct update
+specifications. For more information on using ``Updates`` builder with
+arrays, see our :doc:`guide on the Updates builder </fundamentals/builders/updates>`.
+
+The following example performs these actions: 
+
+- Query for the sample document
+- Add "17" to the ``qty`` array using the ``push()`` method of the ``Updates`` builder
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin pushElementsExample
+   :end-before: end pushElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "color": "green", "qty": [8, 12, 18, 17] }
+
+Specifying Array Elements
+-------------------------
+
+You can specify which array elements to update using a positional
+operator. Positional operators can specify the first, all or certain
+embedded arrays to update. 
+
+To specify elements in an array with positional operators, use the **dot
+notation**. Dot notation is a property access syntax for navigating JSON
+objects. For additional information, see the server manual entry on
+:manual:`dot notation </core/document/#std-label-document-dot-notation>`.
+
+The First Array Element
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To update the first element that matches your query filter, use the
+positional ``$`` operator. 
+
+The positional ``$`` operator specifies the first element of the array
+matched by your query filter.  
+
+Example
+```````
+
+The following example performs these actions:
+
+- Query for the position in ``qty`` where the value is "18"
```

Comment:
```suggestion
- Query for a document with a ``qty`` field containing the value "18"
```

Issue: As the find operation queries for a document with a certain property, I believe it is incorrect to say that the find operation "queries for a position".

Reply:
I disagree. It is looking for a position, hence the use of a positional operator.

Reply:
In synchronous discussion we agreed that the initial comment is correct